src/java/picard/analysis/directed/GcBiasMetricsCollector.java

```diff
@@ -0,0 +1,286 @@
+                    type = this.sample;
+                    group = "Sample";
+                    addRead(gcData.get(type), rec, group, gc, refBases);
+                } else {
+                    type = "All_Reads";
+                    group = "All Reads";
+                    addRead(gcData.get(type), rec, group, gc, refBases);
+                }
+            }
+            else {
+                for (final Map.Entry<String, GcObject> entry : gcData.entrySet()) {
+                    final GcObject gcCur = entry.getValue();
+                    if (!rec.getReadPairedFlag() || rec.getFirstOfPairFlag()) ++gcCur.totalClusters;
+                }
+            }
+        }
+
+        public void finish() {}
+
+        /////////////////////////////////////////////////////////////////////////////
+        // Sums the values in an int[].
+        /////////////////////////////////////////////////////////////////////////////
+        private double sum(final int[] values) {
+            final int length = values.length;
+            double total = 0;
+            for (int i = 0; i < length; ++i) {
+                total += values[i];
+            }
+
+            return total;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////
+        //Called to add metrics to the output file for each level of collection
+        // these metrics are used for graphing gc bias in R script
+        /////////////////////////////////////////////////////////////////////////////
+        public void addMetricsToFile(final MetricsFile<GcBiasMetrics, Integer> file) {
+            for (final Map.Entry<String, GcObject> entry : gcData.entrySet()) {
+                final GcObject gcCur = entry.getValue();
+                final String gcType = entry.getKey();
+                final int[] readsByGc = gcCur.readsByGc;
+                final long[] errorsByGc = gcCur.errorsByGc;
+                final long[] basesByGc = gcCur.basesByGc;
+                final int totalClusters = gcCur.totalClusters;
+                final int totalAlignedReads = gcCur.totalAlignedReads;
+                final String group = gcCur.group;
+                final GcBiasMetrics metrics = new GcBiasMetrics();
+                final double totalWindows = sum(windowsByGc);
+                final double totalReads = sum(readsByGc);
+                final double meanReadsPerWindow = totalReads / totalWindows;
+                if(totalAlignedReads>0) {
+                    for (int i = 0; i < windowsByGc.length; ++i) {
+                        final GcBiasDetailMetrics m = new GcBiasDetailMetrics();
+                        m.GC = i;
+                        m.WINDOWS = windowsByGc[i];
+                        m.READ_STARTS = readsByGc[i];
+                        if (errorsByGc[i] > 0) m.MEAN_BASE_QUALITY = QualityUtil.getPhredScoreFromObsAndErrors(basesByGc[i], errorsByGc[i]);
+                        if(windowsByGc[i]!=0) {
```

Comment:
spacing 'if (windowsByGc[i] != 0) {'
(I know this is just copy and pasting, but it hurts my eyes to see such spacing)